libfreerdp/core/timezone.c

```diff
@@ -76,34 +80,26 @@ void rdp_write_system_time(wStream* s, SYSTEM_TIME* system_time)
 
 BOOL rdp_read_client_time_zone(wStream* s, rdpSettings* settings)
 {
-	char* str = NULL;
-	TIME_ZONE_INFO* clientTimeZone;
+	LPTIME_ZONE_INFORMATION tz;
 
 	if (Stream_GetRemainingLength(s) < 172)
 		return FALSE;
 
-	clientTimeZone = settings->ClientTimeZone;
+	tz = settings->ClientTimeZone;
```

Comment:
Not in the old code but maybe a check for setting->ClientTimeZone makes sense before accessing the members.